---
title: "A taxonomic and molecular survey of the pteridophytes of the Nectandra Cloud Forest Reserve, Costa Rica"
bibliography: "references.bib"
mainfont: Times New Roman.ttf
fontsize: 12pt
mainfontoptions: 
- BoldFont=Times New Roman Bold.ttf
- ItalicFont=Times New Roman Italic.ttf
- BoldItalicFont=Times New Roman Bold Italic.ttf
output: 
  bookdown::pdf_document2:
    citation_package: none # Use pandoc default, otherwise csl can't be used for reference formatting
    toc: no
    number_sections: no
    fig_caption: yes
    keep_tex: no
    latex_engine: xelatex
    pandoc_args: [ "--csl", "plos-one.csl"]
    includes:
      in_header: plos-one.sty
header-includes: 
  \usepackage{float}
  \makeatletter\renewcommand*{\fps@figure}{H}\makeatother
editor_options: 
  chunk_output_type: console
# nocite: | 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE, results = "hide", cache = TRUE)

# Load objects build during analysis from drake cache
loadd(
  list = c(
    "richness_estimate",
    "min_distance_table",
    "specimens",
    "ppgi",
    "cr_richness",
    "DNA_accessions",
    "nectandra_rbcL_raw",
    "iqtree_log"),
  cache = nectandra_cache)

```

```{r taxa-count}
# Count number of taxa at Nectandra
taxa_count <-
  specimens %>%
  left_join(ppgi) %>%
  select(family, genus, species, taxon, specimen) %>%
  unique %>%
  summarize_at(vars(everything()), n_distinct)
```

<!-- Remove page numbers -->
\pagenumbering{gobble}

<!-- Set left justification -->
\raggedright

<!-- Don't indent -->
\setlength{\parindent}{0ex}

Joel H. Nitta^1^*, Atsushi Ebihara^2^, and Alan R. Smith^3^

^1^Department of Botany, National Museum of Natural History, Smithsonian Institution, Washington, District of Columbia, United States of America

^2^Department of Botany, National Museum of Nature and Science, Tsukuba, Ibaraki, Japan  

^3^The University Herbarium, University of California, Berkeley, Berkeley, California, United States of America

*Corresponding author  
E-mail: joelnitta@gmail.com (JHN)

\clearpage

## Abstract

Floristic surveys are crucial to the conservation of biodiversity, but the vast majority of such surveys are limited to listing species names, and few take into account the evolutionary history of species. Here, we combine classical taxonomic and molecular phylogenetic (DNA barcoding) approaches to catalog the biodiversity of pteridophytes (ferns and lycophytes) of the Nectandra Cloud Forest Reserve, Costa Rica. Surveys were carried out over three field seasons (2008, 2011, and 2013), resulting in `r taxa_count %>% pull(species)` species, representing `r taxa_count %>% pull(genus)` genera and `r taxa_count %>% pull(family)` families of pteridophytes. Our literature survey of protected areas in Costa Rica shows that Nectandra has an exceptionally diverse pteridophyte flora for its area. Plastid *rbcL* was selected as a DNA barcode marker and obtained for >95% of pteridophyte taxa at this site. Combined molecular and morphological analysis revealed three previously undescribed taxa that appear to be of hybrid origin. The utility of *rbcL* for species identification was assessed by calculating minimum interspecific distances, and found to have a 14% failure rate. Finally we compared the distribution of minimum interspecific *rbcL* distances with two other areas that have been the focus of pteridophyte molecular surveys, Japan and Tahiti. The comparison shows that Nectandra is more similar to Japan than Tahiti, which may reflect the biogeographic history of these floras.

\clearpage

## Introduction

Despite its small size (\SI{51100}{\square\kilo\meter}), Costa Rica is home to a remarkably high biodiversity, and is ranked as one of the world's top 25 biodiversity hotspots [@Myers2000]. It is estimated that vascular plant species richness in Costa Rica exceeds \num{5000} spp. per \SI{10000}{\square\kilo\meter} [@Barthlott2005]. Accordingly, primary taxonomy is an important part of biological research in Costa Rica for the purposes of both documenting this biodiversity and informing conservation practices. 

In addition to taxonomic diversity, measures of phylogenetic diversity (PD) provide another important perspective on biodiversity and should be taken into consideration for setting conservation priorities [@Faith1992]. Therefore, surveys documenting both species richness and molecular phylogenetic diversity are needed. Furthermore, the use of a standard molecular marker as a DNA "barcode" is useful for species identification and taxonomic revision. For example, DNA barcoding surveys have revealed hidden biodiversity in the form of cryptic species in butterflies in Costa Rica [@Hajibabaei2006], and have been used to identify cryptic life-cycle stages in ferns [@Nitta2017].

Pteridophytes (\ie, ferns and lycophytes) are an important group of plants to study because many new species are still being discovered in the Neotropics [@Almeida2016a] and they play important ecological roles in many ecosystems [\eg, @George1999; @Coomes2005; @Schuettpelz2006c; @Mueller-Dombois2013]. The pteridophytes of Costa Rica comprise \circa \SI{1200}{\species}, accounting for one-quarter to one-third of the estimated richness of the Neotropics (\circa \SIrange{3000}{4500}{\species}) [@Almeida2016a]. While the pteridophyte floras of some sites in Costa Rica are well studied, such as Finca La Selva [@Grayum1987; @oet2019], other areas in the country have received considerably less attention. We present here the results of surveys conducted over three field seasons on the pteridophyte flora of the recently established Nectandra Cloud Forest Reserve near San Ramon, Costa Rica. We also characterize the phylogenetic diversity at this site and compare it with other pteridophyte floras that have recently been the focus of DNA barcoding surveys.

## Materials and Methods

### Study Site

The Nectandra Cloud Forest Reserve (hereafter, "Nectandra"; \ang{10;11;} N, \ang{84;31;} W) is located at \SIrange{1000}{1200}{\meter} a.s.l. on the Atlantic slope of the Cordillera de TilarÃ¡n, Alajuela Province (\fig \@ref(fig:map)a). It encompasses \SI{158}{\hectare} of premontane rainforest [@Holdridge1967], and is an important source of water for the local community [@Lennette2011]. Approximately three-quarters of Nectandra is primary forest with \textgreater98\% canopy cover; the remaining area comprises naturally regenerating former coffee and \textit{Dracaena} plantations (\fig \@ref(fig:map)b, c). Two permanent streams and four seasonal drainages pass through the reserve and empty into the Balsa river. Most of the surrounding land is used for cattle pasture or other agriculture. It has a rich herpetofauna [@Rovito2015] and is home to at least 189 spp. of bryophytes [@Norris2017].

**\fig \@ref(fig:map). Location of study site (Nectandra Cloud Forest Reserve).** (a) Map of Costa Rica, showing location of Nectandra and other protected areas that have been surveyed for pteridophytes. Other protected areas include Alberto Manuel Brenes Biological Reserve, La Selva Biological Station, Monteverde Cloud Forest Reserve, the upper watershed of the Savegre River in the Los Santos Forest Reserve, and San Luis Biological Reserve. (b) Aerial photo of Nectandra (\ca 1995). Nectandra consists of three parcels of land: the original parcel is in white (\ca 80% primary forest), with areas added later in red ("Ocotea" parcel, primary forest) and blue ("Persea" parcel, secondary forest). Note extreme deforestation outside of protected areas. (c) Example of interior of primary forest at Nectandra. Photos courtesy of Evelyne Lennette, used with permission.

Climate at Nectandra is characterized by extremely high frequency of cloud cover throughout the year. Rainfall peaks during the wet season from November to February. Mean annual precipitation is \SIrange{3000}{3500}{\milli\meter\per\year}, with 80\% estimated fog-saturated days.

### Field survey

We carried out surveys of pteridophytes over three field seasons (January 2008, 2011, and 2013; `r specimens %>% filter(!is.na(date_collected)) %>% pull(date_collected) %>% n_distinct` days of sampling total). Most specimens were collected along trails through the reserve. Epiphytes were collected from fallen trees or tree branches, or up to \circa \SI{2}{\meter} on tree trunks. Permits for collection were obtained from the Costa Rican government (SINAC No. 04941 and Cites 2014-CR 1006/SJ (#S 1045)). The top set of voucher specimens were deposited at UC, with duplicates at GH and CR. Hebarium codes follow @Thiers2019. Leaf tissue was preserved on silica gel for DNA extraction. Spores of selected taxa were observed with a standard compound light microscope.

### Taxonomy

We consulted relevant floras for species identification, including *Flora Mesoamericana* [@Moran1995] and @Lellinger1989. Genus-level and higher taxonomy follows Pteridophyte Phylogeny Group I [@PteridophytePhylogenyGroupI2016].

### DNA sequencing and phylogenetic analysis

DNA was extracted with the DNEasy plant mini kit following the manufacturer's protocol (Qiagen). We generally sampled one species per taxon for morphologically distinct taxa, and up to five specimens per taxon for taxa that are more difficult to identify using standard keys [@Moran1995; @Lellinger1989] and morphological characters. We amplified the plastid *rbcL* gene using the PCR primers and thermocycler settings of @Schuettpelz2007. We verified amplification success by gel electrophoresis in 1% TAE. We purified PCR products with Exo-STAR and conducted cycle sequencing using the Big Dye kit with two internal primers, ESRBCL654R and ESRBCL628F [@Schuettpelz2007] in addition to the amplification primers. We imported the resulting AB1 trace files into Geneious [@Kearse2012], assembled contigs, and exported the consensus sequences in FASTA format. *rbcL* sequences were downloaded from GenBank for two taxa that could not be successfully sequenced in our study: *Pteris altissima* Poir. from Nectandra that had been sequenced previously in a different study (*Nitta 863*, KM008147) and *Trichomanes polypodioides* L. from Bolivia (*M. Kessler 8808*, AY175795). We generated an alignment using MAFFT [@Katoh2002], and inferred a tree using IQ-TREE with automatic model selection. Node support was assessed with \num{1000} ultra-fast bootstrap (UFboot) [@Nguyen2015] and \num{1000} ShimodairaâHasegawa-like approximate likelihood ratio test (SH-aLRT) replicates [@Guindon2010] in IQ-TREE. Molecular analysis was authorized by Convenio Marco de CONAGEBIO R-CM-RN-001-2014-OT.

### Statistical analysis

To assess the completeness of sampling, we constructed species incidence rarefaction-extraction curves using the iNEXT package [@Chao2014b] in R v 3.6.0 [@RCoreTeam2019]. Number of collection days was used as the sampling unit.

We compared the number of species at Nectandra with other protected sites in Costa Rica by conducting a literature survey.

We assessed the utility of *rbcL* for species identification by calculating interspecific distances. Species sharing identical *rbcL* sequences with at least one other species (interspecific distance of zero) were considered failures, \ie, not possible to identify with this marker.

To better characterize observed phylogenetic diversity, we compared the phylogenetic diversity of the pteridophytes of Nectandra with two other pteridophyte floras that have been the subject of DNA barcoding using *rbcL*, Japan [@Ebihara2010; @Ebihara2019b] and the islands of Moorea and Tahiti, French Polynesia [@Nitta2017]. To ensure that phylogenetic distances were comparable across datasets, we generated a combined *rbcL* alignment for all species from the three sites together using MAFFT [@Katoh2002]. We then extracted the species for each site and calculated minimum interspecific distances per site.

### Data accessibility

Code to replicate all analyses, figures, and this manuscript are available at https://github.com/joelnitta/nectandra_ferns. Data are available from the Dryad repository at http://doi.org/XXX. Newly generated sequences have been deposited in GenBank (accession numbers XXX--XXX). Photos of voucher specimens available on the Ferns of the World website (https://www.fernsoftheworld.com/florulas/XXX).

\*\*\* NOTE FOR EDITOR AND REVIEWERS: data and code will be made publicly available upon acceptance. During review, data and code can be downloaded from the following link  
https://www.dropbox.com/sh/ojk02yvch8bzz17/AABWCC6Le48p_oV0E_WRF45aa?dl=0. \*\*\*

## Results

### Taxonomic survey

```{r top-taxa}
# Count families and genera with the most species
top_families <-
  specimens %>%
  left_join(ppgi) %>%
  select(family, genus, species) %>%
  unique %>%
  group_by(family) %>%
  summarize(n = n()) %>%
  arrange(desc(n))

top_genera <-
  specimens %>%
  left_join(ppgi) %>%
  select(family, genus, species) %>%
  unique %>%
  group_by(genus) %>%
  summarize(n = n()) %>%
  arrange(desc(n))

nectandra_sp_per_hec <-
cr_richness %>%
  mutate(sp_per_hectare = richness / area_ha) %>%
  arrange(desc(sp_per_hectare)) %>%
  slice(1) %>%
  pull(sp_per_hectare) %>%
  round(2)

# Obtain asymptotic estimated species richness
est_rich <-
richness_estimate[["AsyEst"]] %>%
  as.data.frame %>%
  rownames_to_column("metric") %>%
  as_tibble %>%
  janitor::clean_names() %>%
  filter(metric == "Species Richness") %>%
  mutate_if(is.numeric, round)

```

Our surveys resulted in `r taxa_count %>% pull(specimen)` individuals, representing `r taxa_count %>% pull(species)` species, `r taxa_count %>% pull(genus)` genera, and `r taxa_count %>% pull(family)` families of pteridophytes (\tab S1).

The genera with the most species include *`r top_genera %>% slice(1) %>% pull(genus)`* (`r top_genera %>% slice(1) %>% pull(n)` spp.), *`r top_genera %>% slice(2) %>% pull(genus)`* (`r top_genera %>% slice(2) %>% pull(n)` spp.), *`r top_genera %>% slice(3) %>% pull(genus)`* (`r top_genera %>% slice(3) %>% pull(n)` spp.), and *`r top_genera %>% slice(4) %>% pull(genus)`* (`r top_genera %>% slice(4) %>% pull(n)` spp.). Families with the most species include `r top_families %>% slice(1) %>% pull(family)` (`r top_families %>% slice(1) %>% pull(n)` spp.), `r top_families %>% slice(2) %>% pull(family)` (`r top_families %>% slice(2) %>% pull(n)` spp.), and `r top_families %>% slice(3) %>% pull(family)` (`r top_families %>% slice(3) %>% pull(n)` spp.).

Three taxa could not be matched to any known species, *Polyphlebium* sp1 (*Nitta 123* and *Nitta 2378*), *Campyloneurum* sp1 (*Nitta 2308*), and *Megalastrum* sp1 (*Nitta 727*).

Our literature survey identified five other protected areas in mainland Costa Rica that have been surveyed for pteridophytes [@oet2019; @Haber1990; Kappelle2000; @Rojas2009; @Jimenez2016] (\fig \@ref(fig:map)). The site with the highest species richness is the Alberto Manuel Brenes Biological Reserve (281 species), with Nectandra the second-highest and nearly the same as La Selva Biological Station (\fig \@ref(fig:richness-cr)a). Nectandra has by far the most species per hectare (`r nectandra_sp_per_hec`), with the San Luis Biological Reserve second-highest (\fig \@ref(fig:richness-cr)b).

**\fig \@ref(fig:richness-cr). Species richness of pteridophytes at protected areas in Costa Rica.**  
For full names of protected areas, \fig \@ref(fig:map). (a) number of species. (b) number of species per hectare.

The collection curve did not reach an asymptote (\fig \@ref(fig:richness-inext)). Extrapolation of the curve indicates that the true species richness may approach `r est_rich$estimator` spp. (95% confidence interval `r est_rich$x95_percent_lower`--`r est_rich$x95_percent_upper` spp.).

**\fig \@ref(fig:richness-inext). Interpolation (solid line) and extrapolation (dashed line) of species richness of pteridophytes at Nectandra Cloud Forest Reserve, Costa Rica.**  
Point at transition from solid to dashed line indicates observed richness. Grey shading indicates 95% confidence interval.

### Phylogenetic analysis

```{r seq-stats}
# Make table of newly generated sequences that will be submitted to GenBank
new_seqs_data <-
tibble(genomic_id = names(nectandra_rbcL_raw)) %>%
  left_join(DNA_accessions, by = "genomic_id") %>%
  left_join(specimens, by = "specimen_id") %>%
  assert(not_na, taxon)

# Check number of new rbcL sequences
n_taxa_rbcL_seqs <-
new_seqs_data %>% pull(taxon) %>% n_distinct()

# Check coverage of rbcL sequences
coverage_taxa_rbcL_seqs <- 
  n_taxa_rbcL_seqs %>%
  magrittr::divide_by(taxa_count$taxon) %>%
  round(3) %>%
  scales::percent()

# Extract relevant lines from iqtree log
align_stats <- 
  magrittr::extract(
    iqtree_log, 
    iqtree_log %>% str_detect("Alignment has [:digit:]+ sequences"))

model_stats <-
  magrittr::extract(
    iqtree_log, 
    iqtree_log %>% str_detect("Best-fit model: "))

best_mod <-
  model_stats %>%
  str_match("model: (.+) chosen") %>%
  magrittr::extract(,2)

best_mod_basis <-
  model_stats %>%
  str_match("according to (.+)$") %>%
  magrittr::extract(,2)

# Get vector of sequence lengths
seq_lengths <-
nectandra_rbcL_raw %>% map_dbl(length)

```

We recovered *rbcL* sequences for `r n_taxa_rbcL_seqs` taxa (`r coverage_taxa_rbcL_seqs`) from Nectandra, resulting in `r new_seqs_data$genomic_id %>% n_distinct` new *rbcL* sequences total. Mean sequence length was `r mean(seq_lengths) %>% round(1)` Â± SD `r sd(seq_lengths) %>% round(1)` bp. *rbcL* could not be sequenced for `r taxa_count$taxon - n_taxa_rbcL_seqs` taxa (of these, one already had a sequence available on GenBank), and only partial sequences (ca. 3' or 5' half of *rbcL*) could be obtained for `r seq_lengths[seq_lengths < 1000] %>% length()` taxa. Difficulty in sequencing of these samples may be due to degraded condition of DNA.

The *rbcL* alignment was `r get_number(align_stats, "columns")` bp long including `r get_number(align_stats, "informative sites")` parsimony-informative sites and `r get_number(align_stats, "sequences")` sequences. The `r best_mod` model was selected for ML phylogenetic analysis according to `r best_mod_basis`.

The *rbcL* phylogeny (\fig S1) was in generally good agreement with recently published plastid phylogenies [@Pryer2004a; @Schuettpelz2007; @Lehtonen2011; @Testo2016a] at the family level and above, with families *sensu* PPGI [@PteridophytePhylogenyGroupI2016] were monophyletic. One exception to Osmundaceae sister to Gleicheniaceae (SH-aLRT < 50%; UFboot 95%), which may be due to poor sampling; we could only obtain a 614 bp *rbcL* fragment from the sole osmundaceous species at our study site. Some internal nodes had weak support (\eg, Cyatheales [SH-aLRT 66%; UFboot 93%], leptosporangiates [SH-aLRT < 50%; UFboot 87%]). It should be noted that IQ-TREE UFboot support values are not analogous to traditional bootstrap values; only nodes receiving SH-aLRT >= 80% and UFboot >= 95% should be considered reliable [@Minh2013]. As the purpose of this study was not to robustly infer phylogeny across the ferns and lycophytes, we do not discuss such deep relationships further.

A small number of genera were found to be non-monophyletic but lacked strong support. *Stenogrammitis limula* (Christ) Labiak is nested within *Lellingeria* (SH-aLRT 87%; UFboot 99%). *Radiovittaria minima* (Baker) E.H. Crane is sister to a clade including *Radiovittaria remota* (FÃ©e) E.H. Crane, *Polytaenium lineatum* (Sw.) J. Sm., and *Vittaria bradeorum* Rosenst. (SH-aLRT 84%; UFboot 75%). *Sphaeropteris* is nested within *Cyathea* (SH-aLRT 93%; UFboot 95%) as the sister to *Cyathea bicrenata* Liebm. (SH-aLRT 76%; UFboot 93%).

Sampling of multiple specimens per species for taxa that are morphologically difficult to resolve revealed several non-monophyletic species, indicative of species complexes. *Megalastrum apicale* R.C. Moran & J. Prado, *M. atrogriseum* (C. Chr.) A.R. Sm. & R.C. Moran, *M. longipilosum* A. Rojas, are closely related (SH-aLRT 93%; UFboot 99%) and the monophyly of each lacks support. *Diplazium carnosum* Christ is non-monophyletic with respect to *D. urticifolium* Christ and *D. macrophyllum* Desv., but this and most other relationships within *Diplazium* lacked support. *Didymoglossum ekmanii* (Wess. Boer) Ebihara & Dubuisson is nested within, and very closely related to *Didymoglossum kapplerianum* (J.W. Sturm) Ebihara & Dubuisson (SH-aLRT 100%; UFboot 100%).

### Barcode analysis

The pteridophyte flora of Nectandra includes `r min_distance_table %>% filter(dataset == "CR", is_zero == TRUE) %>% pull(n)` taxa (`r  min_distance_table %>% filter(dataset == "CR", is_zero == TRUE) %>% pull(percent) %>% percent`) that share identical *rbcL* sequences with at least one other taxon. This failure rate is higher than that of the pteridophytes of Moorea and Tahiti, but lower than Japan (\fig \@ref(fig:min-dist)).

**\fig \@ref(fig:min-dist). Minimum interspecific distances for selected pteridophyte floras**  
(a) Nectandra, (b) Moorea and Tahiti, (c) Japan (all species), and (d) Japan (sexual diploids only). Red bar indicates interspecific distance of zero, i.e., species that cannot be distinguished using *rbcL*.

## Discussion

Here, we present the first combined taxonomic and molecular survey of pteridophytes in Costa Rica. We place our results in regional and global context by comparing this flora with other protected areas in Costa Rica and two other sites that have been the focus of DNA barcoding, Tahiti and Japan.

### Taxonomic diversity and species richness

Nectandra has the second-highest species richness of protected areas in Costa Rica with data available for ferns and lycophytes, and by far the greatest number of species per hectare. While the number of species per hectare is not a fair measure of biodiversity per se as the species-area curve is not linear [@Arrhenius1921], it is useful to assess the effectiveness of a given protected area. Clearly, Nectandra is highly effective at protecting a large number of pteridophyte species given its area. Furthermore, the collection curve indicates that additional, unsampled species may be present (\fig \@ref(fig:richness-inext)), adding to the value of this conservation area. A recent survey of the bryophytes of Nectandra found a similar number of species (189) and also suggested unsampled species remained due to the shape of the collection curve [@Norris2017]. One reason for the high species richness at Nectandra may be its elevation (\SIrange{1000}{1200}{\meter}). Species richness of pteridophytes in the tropics generally reaches a maximum at mid-elevations on mountains [@Kessler2011a], and plot-based surveys of pteridophytes along elevational gradients in Costa Rica spanning \circa \SIrange{100}{3000}{\meter} have found maximum richness at \SIrange{1000}{1200}{\meter} [@WatkinsJr.2006a; @Karger2011].

### Unidentified taxa and species complexes

Of the unidentified taxa, two are likely hybrids between distinct species. *Polyphlebium* sp1 (*Nitta 123* and *Nitta 2378*) shares very similar *rbcL* sequences with *Polyphlebium capillaceum*  (L.) Ebihara & Dubuisson but differs from this species by having expanded laminae (vs. laminae reduced to a few cells on either side of the veins) and growing on rocks in stream beds (vs. growing epiphytically on tree ferns). *Polyphlebium* sp1 has 32 spores per sporangium, a condition that often indicates asexual reproduction via apogamy [@Braithwaite1964]. This may allow it to reproduce by spores despite not being able to complete normal meiosis. To our knowledge, a count of 32 spores per sporangium has not been previously reported from *Polyphlebium*, and more detailed study is needed to confirm the reproductive mode of this taxon. *Campyloneurum* sp1 (*Nitta 2308*) matches in *rbcL* exactly with *Campyloneurum angustifolium* (Sw.) FÃ©e, but has narrower fronds. *Campyloneurum* sp1 has clear, misshappen spores, whereas *C. angustifolium* is a sexual diploid with oblong spores (2*n* = 72; @Leon1992). It is likely that each of these taxa is a hybrid between the species with they share *rbcL* as the mother and another unknown species as the father.

*Megalastrum* sp1 (*Nitta 727*) shares common aspects of morphology with *M. longipilosum*, but the *rbcL* sequences of *M. apicale*, *M. atrogriseum*, *M. longipilosum* are extremely similar. This group is likely a species complex and needs further study, in particular with regards to ploidy level and reproductive mode. Similarly, *Diplazium* also showed a high degree of morphological diversity but extremely low divergence in *rbcL* sequences, and at least one species, *D. carnosum*, appears to be non-monophyletic. *Diplazium* at Nectandra may also be a species complex including multiple hybrids and parental taxa. Sequencing of nuclear genes is needed to clarify relationships for both the putative hybrids (*Polyphlebium* sp1 and *Campyloneurum* sp1) as well as these species complexes.

### Molecular diversity and DNA barcode suitability

Unlike animals, there is no single DNA barcode available for plants that works to reliably identify species across all taxonomic groups [@Group2008]. *rbcL* has relatively high phylogenetic informativeness in ferns and lycophytes at the species level and is one of the most frequently sequenced plastid genes in molecular systematic studies. In previous comparisons with other common plant barcode markers in pteridophytes, *rbcL* performs better than *trnH-psbA* and comparably with *matK* [@Li2011; @Nitta2017]. However, universal primers for *matK* in pteridophytes are lacking [@Li2011]. *rbcL* therefore is a reasonable choice for a single barcode marker in pteridophytes. However, there have been relatively few studies comparing the performance of *rbcL* across different pteridophyte floras.

We find that the pteridophytes of Nectandra have a species identification failure rate intermediate between that of Moorea and Tahiti (French Polynesia) and Japan. The distribution of minimum interspecific distances may reflect the biogeographic history of each region. Costa Rica and Japan are both mainland areas (or formerly connected with the mainland in the case of Japan), whereas the islands of French Polynesia are extremely isolated and have never been connected to a continent. It is likely that more species in French Polynesia are recent immigrants that have mostly evolved elsewhere, compared to more species that have evolved *in situ* in Costa Rica and Japan. This would result in the observed distribution of high interspecific divergences for species from French Polynesia and low interspecific divergences for species in Costa Rica and Japan.

Furthermore, the high failure rate in Japan is likely due to the high rate of apogamy combined with the taxonomic practice of splitting apogamous and sexual forms into separate species [@Ebihara2019b]. When only sexual diploids are included, the failure rate in Japan drops and the distribution more closely resembles that of Costa Rica (\fig \@ref(fig:min-dist)).

## Conclusions

The Nectandra Cloud Forest Reserve is clearly an important site for biodiversity of ferns and lycophytes in Costa Rica, harboring a high number of species for its relatively small area. Our surveys have already revealed multiple taxa that appear to be new to science, and more likely remain to be discovered. We demonstrated that *rbcL* can be used to reliably distinguish species in \circa 85% of cases at Nectandra. This will enable future studies to investigate the ecology of these pteridophyte communities in more depth, particularly with regards to the gametophytic phase. Other studies using *rbcL* as a DNA barcode to identify fern gametophytes to species have revealed particular clades and morphologies that tend to occur long distances from conspecific sporophytes [@Ebihara2013; @Nitta2017], but such patterns have yet to be tested in the Neotropics. We hope our study will lead to more ecological research incorporating field observations of gametophytes and contribute to the systematics of neotropical pteridophytes.

## Acknowledgments

The authors are very grateful to Evelyne and David Lennette for allowing us to study the pteridophytes of Nectandra over multiple field seasons and for providing logistical and financial support. Freddy Castillo provided assistance with fieldwork. This study was supported by the Japan Society for the Promotion of Science (Kakenhi grant no. 15K07204).

## References
<!-- Insert <div id="refs"></div> to place the refs here instead of at the end -->
<div id="refs"></div>

\clearpage

```{r map, fig.height = 4, fig.width = 5, fig.cap="Map of Costa Rica"}

# Download altitude data for Costa Rica
dem.raster <- getData("alt", country = "CR")
dem.m  <-  rasterToPoints(dem.raster)
dem.df <-  data.frame(dem.m)
colnames(dem.df) = c("lon", "lat", "alt")

# Make Costa Rica map showing location of conservation areas
costa_rica_map <-
ggplot() +
  geom_raster(data = dem.df, aes(lon, lat, fill = alt)) +
  scale_fill_scico(palette = "nuuk") +
  geom_point(data = cr_richness, aes(x =  longitude, y = latitude, shape = name), size = 2) +
  labs(
    fill = "Elevation (m)",
    shape = "Site"
  ) +
  scale_y_continuous(labels = scales::degree_format(unit = "N", accuracy = 1)) +
  scale_x_continuous(labels = scales::degree_format(unit = "W", accuracy = 1, scale = -1)) +
  coord_quickmap() +
  standard_theme3() +
  theme(
    axis.title.x = element_blank(),
    axis.title.y = element_blank()
  )

# Load Nectandra map created with Inkscape
nectandra_map <- ggdraw() + draw_image("images/nectandra_property_map.png")

# Load primary forest photo
photo <- ggdraw() + draw_image("images/P1070534.jpg")

# Create a blank spacer plot to add some white space
# between Nectandra map and photo
blank <- patchwork::plot_spacer()

# The top part of the plot is just the Costa Rica map
top <- plot_grid(costa_rica_map, labels = "a")

# The bottom is Nectandra map and photo
bottom <- plot_grid(nectandra_map, blank, photo, nrow = 1, labels = c("b", "", "c"), rel_widths = c(54.75, 0.5, 54.75))

# Put them together
plot_grid(top, bottom, nrow = 2, ncol = 1)

# Save the file for submission
ggsave(filename = "Fig1.tiff", height = 7.25, width = 7.5, units = "in", dpi = 350)
```

\clearpage

```{r richness-cr, fig.height=3, fig.width=5, fig.cap="Species richness of pteridophytes at protected areas"}

a <-
  cr_richness %>%
  mutate(name = fct_reorder(name, richness)) %>%
  ggplot(aes(x = name, y = richness)) +
  geom_col() +
  standard_theme3() +
  coord_flip() +
  labs(y = "No. spp.", x = "Site", subtitle = "a")

b <-
  cr_richness %>%
  mutate(sp_per_hectare = richness / area_ha) %>%
  mutate(name = fct_reorder(name, sp_per_hectare)) %>%
  ggplot(aes(x = name, y = sp_per_hectare)) +
  geom_col() +
  standard_theme3() +
  coord_flip() +
  labs(y = "No. spp. per hectare", x = "", subtitle = "b")

a + b

ggsave(filename = "Fig2.eps", height = 3, width = 5, units = "in")

```

\clearpage

```{r richness-inext, fig.height=3, fig.width=3, fig.cap="Estimated richness"}
make_inext_plot(richness_estimate)

ggsave(filename = "Fig3.eps", height = 3, width = 3, units = "in")

```

\clearpage

```{r min-dist, fig.height=4.5, fig.width=7, fig.cap="Minimum interspecific distances"}

min_distance_table %>%
  mutate(dataset = case_when(
    dataset == "CR" ~ "a",
    dataset == "FP" ~ "b",
    dataset == "JA" ~ "c",
    dataset == "JAsexdip" ~ "d"
  )) %>%
  ggplot(aes(range, percent, fill = is_zero)) +
  geom_col() +
  scale_fill_manual(values = c("dark grey", "red")) +
  scale_x_continuous(labels = scales::percent) +
  scale_y_continuous(labels = scales::percent) +
  standard_theme2() +
  theme(legend.position = "none") +
  facet_wrap(vars(dataset), ncol = 1) +
  labs(x = "Genetic distance", y = "Relative frequency") +
  theme(
    strip.text = element_text(hjust = 0),
    strip.background = element_blank())

ggsave(filename = "Fig4.eps", height = 4.5, width = 7, units = "in")

```

\clearpage

## Supporting Information

Table S1. A checklist of fern and lycophyte species observed at the Nectandra Cloud Forest Reserve, Costa Rica. All voucher specimens deposited in UC.

Table S2. GenBank accession numbers of sequences analyzed in this study.

Figure S1. Maximum-likelihood phylogenetic tree of ferns and lycophytes at the Nectandra Cloud Forest Reserve, Costa Rica inferred using IQ-TREE with \num{1000} SH-like approximate likelihood ratio test (SH-aLRT) and ultra-fast bootstrap (UFboot) replicates each. Tree rooted on lycophytes. Numbers at nodes indicate SH-aLRT support (%) / UFboot support (%); values less than 50% shown with "-"; values of 100% shown with "\*". For phylogram on right side, scale bar shows expected number of changes per site. Numbers after species name are J. H. Nitta specimen collection numbers of specimens from Nectandra, except for *Trichomanes polypodiodes*, which not be sequenced successfully, so a GenBank sequence (AY175795) from Bolivia (*M. Kessler 8808*) was used instead.
