---
title: "A taxonomic and molecular survey of the pteridophytes of the Nectandra Cloud Forest Reserve, Costa Rica"
bibliography: 
  - "references.bib"
  - "references_other.yaml"
mainfont: Roboto-Regular.ttf
fontsize: 12pt
mainfontoptions: 
- BoldFont=Roboto-Bold.ttf
- ItalicFont=Roboto-Italic.ttf
- BoldItalicFont=Roboto-BoldItalic.ttf
output: 
  bookdown::pdf_document2:
    citation_package: none # Use pandoc default, otherwise csl can't be used for reference formatting
    toc: no
    number_sections: no
    fig_caption: yes
    keep_tex: yes
    latex_engine: xelatex
    pandoc_args: [ "--csl", "plos-one.csl"]
    includes:
      in_header: plos-one.sty
header-includes: 
  \usepackage{float}
  \makeatletter\renewcommand*{\fps@figure}{H}\makeatother
editor_options: 
  chunk_output_type: console
# nocite: | 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE, results = "hide", cache = FALSE)

# Load objects build during analysis from drake cache
loadd(
  list = c(
    "checklist",
    "richness_estimate",
    "min_distance_table",
    "nectandra_specimens",
    "nectandra_specimens_strict",
    "ppgi",
    "cr_richness",
    "costa_rica_el",
    "nectandra_dna",
    "nectandra_rbcL",
    "nectandra_rbcL_raw",
    "nectandra_rbcL_raw_renamed",
    "nectandra_rbcL_missing_taxa",
    "iqtree_log",
    "rbcL_tree"),
  cache = nectandra_cache)

# Define helper functions:

# Extract the top genus and families with most species
# use ... to rank by 'genus' or 'family'
top_taxon <- function (df, rank, ...) {
  df %>% arrange(desc(n)) %>% slice(rank) %>% pull(...)
}

top_taxon_rich <- function (df, rank) {
  df %>% arrange(desc(n)) %>% slice(rank) %>% pull(n) %>% si("spp.")
}

# Extract the number and percentage
# of barcode failures by dataset

barcode_fail_num <- function(df, dataset_select) {
  df %>% filter(dataset == dataset_select, is_zero == TRUE) %>% pull(n)
}

barcode_fail_percent <- function(df, dataset_select) {
  df %>% filter(dataset == dataset_select, is_zero == TRUE) %>% pull(percent) %>% percent
}

```

<!-- Set left justification -->
\raggedright

<!-- Don't indent -->
\setlength{\parindent}{0ex}

Joel H. Nitta^1^*, Atsushi Ebihara^2^, and Alan R. Smith^3^

^1^Department of Biological Sciences, Graduate School of Science, University of Tokyo, Bunkyo-ku, Tokyo, Japan

^2^Department of Botany, National Museum of Nature and Science, Tsukuba, Ibaraki, Japan

^3^The University Herbarium, University of California at Berkeley, Berkeley, California, United States of America

*Corresponding author  
E-mail: \textnormal{joelnitta@gmail.com} (JHN)

\clearpage

## Abstract

```{r abstract-stats}
# Count number of taxa at Nectandra
taxa_count <-
  nectandra_specimens_strict %>%
  left_join(ppgi, by = "genus") %>%
  select(family, genus, species, taxon, specimen) %>%
  assert(not_na, family, genus, species, taxon, specimen) %>%
  assert(is_uniq, specimen) %>%
  summarize_at(vars(everything()), n_distinct)

# Count the percentage of taxa with DNA sequences from Nectandra
percent_taxa_with_dna_from_nectandra <-
  # Start with aligned rbcL including GenBank sequences
  tibble(tip = rownames(nectandra_rbcL)) %>%
  # Check that there are two non-Nitta samples
  # (these are not from Nectandra and will be excluded)
  mutate(not_nitta_sample = str_detect(tip, "Nitta", negate = TRUE)) %>%
  verify(sum(not_nitta_sample) == 2) %>%
  filter(!not_nitta_sample) %>%
  # Extract Nitta specimen collection number
  mutate(
    specimen = str_match(tip, "_(Nitta_.+)$") %>% 
      magrittr::extract(,2) %>%
      str_replace_all("_", " ")) %>%
  assert(not_na, specimen) %>%
  assert(is_uniq, specimen) %>%
  # Inner-join with only specimens from Nectandra to filter
  inner_join(nectandra_specimens_strict, by = "specimen") %>%
  assert(not_na, taxon) %>%
  # Count distinct taxa
  pull(taxon) %>%
  n_distinct() %>%
  # Convert to percentage of all taxa at Nectandra
  magrittr::divide_by(n_distinct(nectandra_specimens_strict$taxon)) %>%
  # Round down to nearest integer
  magrittr::multiply_by(100) %>%
  floor() %>%
  magrittr::divide_by(100) %>%
  scales::percent()
```

Floristic surveys are crucial to the conservation of biodiversity, but the vast majority of such surveys are limited to listing species names, and few take into account the evolutionary history of species.
Here, we combine classical taxonomic and molecular phylogenetic (DNA barcoding) approaches to catalog the biodiversity of pteridophytes (ferns and lycophytes) of the Nectandra Cloud Forest Reserve, Costa Rica.
Surveys were carried out over three field seasons (2008, 2011, and 2013), resulting in `r taxa_count %>% pull(species)` species representing `r taxa_count %>% pull(genus)` genera and `r taxa_count %>% pull(family)` families of pteridophytes.
Our literature survey of protected areas in Costa Rica shows that Nectandra has an exceptionally diverse pteridophyte flora for its size.
Plastid *rbcL* was selected as a DNA barcode marker and obtained for >`r percent_taxa_with_dna_from_nectandra` of pteridophyte taxa at this site.
Combined molecular and morphological analyses revealed three previously undescribed taxa that appear to be of hybrid origin.
The utility of *rbcL* for species identification was assessed by calculating minimum interspecific distances and found to have a failure rate of `r barcode_fail_percent(min_distance_table, "CR")`.
Finally we compared the distribution of minimum interspecific *rbcL* distances with two other areas that have been the focus of pteridophyte molecular surveys: Japan and Tahiti.
The comparison shows that Nectandra is more similar to Japan than Tahiti, which may reflect the biogeographic history of these floras.

\clearpage

## Introduction

Despite its small area (`r si(51100, "km^2^")`), Costa Rica is home to remarkably high biodiversity, and is ranked as one of the world's top 25 biodiversity hotspots [@Myers2000].
It is estimated that vascular plant species richness in Costa Rica exceeds `r si(5000, "spp.")` per `r si(10000, "km^2^")` [@Barthlott2005].
Accordingly, primary taxonomy is an important part of biological research in Costa Rica for the purposes of both documenting this biodiversity and informing conservation practices. 

In addition to taxonomic diversity, measures of phylogenetic diversity (PD) provide an important perspective on biodiversity and should be taken into consideration for setting conservation priorities [@Faith1992].
Therefore, surveys documenting both species richness and molecular phylogenetic diversity are needed.
Furthermore, the use of a standard molecular marker as a DNA "barcode" is useful for species identification and taxonomic revision [@Hebert2003b].
For example, DNA barcoding surveys have revealed hidden biodiversity in the form of cryptic species in butterflies in Costa Rica [@Hajibabaei2006] and have been used to identify cryptic life-cycle stages in ferns [@Nitta2017; @Ebihara2013]. 
DNA barcodes are being increasingly integrated into biological surveys to document biodiversity at unprecedented scale, rate, and resolution [\eg, @Janzen2016; @Delrieu-Trottin2019; @Telfer2015].

Pteridophytes (\ie, ferns and lycophytes) are an important group of plants to study because many new species are still being discovered in the Neotropics [@Almeida2016a] and they play important ecological roles in many ecosystems [\eg, @George1999; @Coomes2005; @Schuettpelz2006c; @Mueller-Dombois2013].
The pteridophytes of Costa Rica comprise \circa{ }`r si(1200, "spp.")`, accounting for one-quarter to one-third of the estimated richness of the Neotropics (\circa{ }`r si(3000, "spp.")` to `r si(4500, "spp.")`) [@Almeida2016a].
While the pteridophyte floras of some sites in Costa Rica are well studied, such as La Selva Biological Station [@Grayum1987; @oet2019], other areas in the country have received considerably less attention.
We present here the results of surveys conducted over three field seasons on the pteridophyte flora of the recently established Nectandra Cloud Forest Reserve near San Ramon, Costa Rica.
We also characterize the phylogenetic diversity at this site and compare it with other pteridophyte floras that have recently been the focus of DNA barcoding surveys.

## Materials and methods

### Study site

The Nectandra Cloud Forest Reserve (hereafter, "Nectandra"; 10°11' N, 84°31' W) is located at `r si(1000, "m")` to `r si(1200, "m")` a.s.l. on the Atlantic slope of the Cordillera de Tilarán, Alajuela Province (`r figure("map")`a).
It encompasses `r si(158, "ha")` of premontane rainforest (life zones follow Holdridge [@Holdridge1967]) and is an important source of water for the local community [@Lennette2011].
Approximately three-quarters of Nectandra is primary forest with >98% canopy cover; the remaining area comprises naturally regenerating former coffee and *Dracaena* plantations (`r figure("map")`b, c).
Two permanent streams and four seasonal drainages pass through the reserve and empty into the Balsa river.
Most of the surrounding land is used for cattle pasture or other agriculture.
It has a rich herpetofauna [@Rovito2015] and is home to at least `r si(188, "spp.")` of bryophytes [@Norris2017].

**`r figure_full("map")`** (a) Map of Costa Rica, showing location of Nectandra (yellow diamond) and other protected areas that have been surveyed for pteridophytes (black dots).
Other protected areas include Alberto Manuel Brenes Biological Reserve, La Selva Biological Station, Monteverde Cloud Forest Reserve, San Luis Biological Reserve, and the upper watershed of the Savegre River in the Los Santos Forest Reserve.
(b) Aerial photo of Nectandra (\circa{ }1995).
Nectandra consists of three parcels of land: the original parcel is in white (\circa{ }80% primary forest), with areas added later in red ("Ocotea" parcel, primary forest) and blue ("Persea" parcel, secondary forest).
Note extreme deforestation outside of protected areas.
(c) Example of interior of primary forest at Nectandra.
Photos courtesy of Evelyne Lennette, used with permission.

Climate at Nectandra is characterized by extremely high frequency of cloud cover throughout the year.
Rainfall peaks during the wet season from November to February.
Mean annual precipitation is `r si(3000, "mm yr^-1^")` to `r si(3500, "mm yr^-1^")`, with \circa{ }80% fog-saturated days.

### Field survey

We carried out surveys of pteridophytes over three field seasons (January 2008, 2011, and 2013; `r nectandra_specimens_strict %>% filter(!is.na(date_collected)) %>% pull(date_collected) %>% n_distinct` days of sampling total).
Most specimens were collected along trails through the reserve.
Epiphytes were collected from fallen trees or tree branches, or up to \circa{ }`r si(2, "m")` on tree trunks.
Permits for collection were obtained from the Costa Rican government (SINAC No. 04941 and Cites 2014-CR 1006/SJ [#S 1045]).
The first set of voucher specimens was deposited at UC, with duplicates at CR, GH, TNS, and the private collection at Nectandra.
Herbarium codes follow Thiers [@Thiers2020].
Leaf tissue was preserved on silica gel for DNA extraction.
Spores of selected taxa were observed with a standard compound light microscope.

### Taxonomy

We consulted relevant floras and monographs for species identification [@Moran1995; @Lellinger1989; @Mickel2004; @Hirai2011; @Moran2010b; @Sundue2014a; @Gonzales-Rocabado2011; @Kessler2011b; @Moran2018a; @Rojas-Alvarado2017a].
Genus-level and higher taxonomy follows Pteridophyte Phylogeny Group I [@PteridophytePhylogenyGroupI2016].

### DNA sequencing and phylogenetic analysis

DNA was extracted with the DNEasy Plant Mini kit following the manufacturer's protocol (Qiagen).
We generally sampled one species per taxon from Nectandra for morphologically distinct taxa, and up to five specimens per taxon for taxa that are more difficult to identify using standard keys [@Moran1995; @Lellinger1989] (\eg, *Megalastrum*, *Didymoglossum*).
In a small number of cases, we used samples from other areas in Costa Rica if the sample from Nectandra could not be successfully sequenced.
We selected the plastid *rbcL* gene as a barcode marker because it has universal primers available for ferns [@Schuettpelz2007] and has performed relatively well for species identification in ferns relative to other candidate barcode loci [@Li2011; @Ebihara2010; @Nitta2017].
We amplified *rbcL* using PCR primers and thermocycler settings of @Schuettpelz2007 and verified amplification success by gel electrophoresis in 1% TAE.
We purified PCR products with Exo-STAR (GE Healthcare) and conducted cycle sequencing using the Big Dye Terminator v3.1 Cycle Sequencing Kit (ThermoFisher) with two internal primers, ESRBCL654R and ESRBCL628F [@Schuettpelz2007], in addition to the amplification primers.
We imported the resulting AB1 trace files into Geneious [@Kearse2012], assembled contigs, and exported the consensus sequences in FASTA format.
We downloaded *rbcL* sequences from GenBank if available for any remaining taxa that could not be successfully sequenced (`r s_table("genbank")`).
We generated an alignment using MAFFT [@Katoh2002].
For phylogenetic analysis by maximum likelihood, all sites were included in a single partition (no partitioning was specified).
We evaluated models of DNA sequence evolution with IQ-TREE [@Nguyen2015] ("-m TEST"), which is similar to the model selection process implemented in jModelTest [@Posada2008].
The Bayesian Information Criterion (BIC) was used to select the best model (IQ-TREE default setting), which was then used by IQ-TREE to infer the tree.
Node support was assessed with `r si(1000)` ultra-fast bootstrap (UFboot) [@Nguyen2015] and `r si(1000)` Shimodaira-Hasegawa-like approximate likelihood ratio test (SH-aLRT) replicates [@Guindon2010] in IQ-TREE.
For a small number of genera that were not supported as monophyletic in the original phylogenetic analysis, we also downloaded all available *rbcL* sequences for closely related taxa (at the family or subfamily level) from GenBank, aligned these in combination with the newly generated sequences from Nectandra with MAFFT, and inferred a tree using FastTree on default settings [@Price2009; @Price2010].
Molecular analysis was authorized by Convenio Marco de CONAGEBIO R-CM-RN-001-2014-OT.

### Statistical analysis

To assess the completeness of sampling, we constructed species incidence rarefaction-extraction curves using the iNEXT package [@Chao2014b].
Number of collection days was used as the sampling unit.

We compared the number of species at Nectandra with other protected sites in Costa Rica by conducting a literature survey.

We assessed the utility of *rbcL* for species identification by calculating minimum interspecific distances as follows.
We first calculated all raw interspecific distances in the *rbcL* alignment using the "dist.dna" function in the APE package [@Paradis2004], then extracted the minimum interspecific distance for each species using custom scripts.
Species sharing identical *rbcL* sequences with at least one other species (interspecific distance of zero) were considered failures, \ie, not possible to identify with this marker.

To better characterize observed phylogenetic diversity, we compared the phylogenetic diversity of the pteridophytes of Nectandra with two other pteridophyte floras that have been the subject of DNA barcoding using *rbcL*: Japan [@Ebihara2010; @Ebihara2019b] and the islands of Moorea and Tahiti, French Polynesia [@Nitta2017].
To ensure that phylogenetic distances were comparable across datasets, we generated a combined *rbcL* alignment for all species from the three floras together using MAFFT.
We then subset the alignment to the species in each flora and calculated minimum interspecific distances per flora as described above.

All analyses were carried out using R v 3.6.1 [@RCoreTeam2019].

### Data accessibility

Code to replicate all analyses, figures, and this manuscript is available at https://github.com/joelnitta/nectandra_ferns.
A Docker image is available to run the code at https://hub.docker.com/r/joelnitta/nectandra_ferns.
Data are available from the Dryad repository [@Nitta2020b].
Newly generated sequences have been deposited in GenBank (`r s_table("genbank")`).
Photos of voucher specimens are available on the Ferns of the World website (https://www.fernsoftheworld.com/florulas/XXX).

\*\*\* NOTE FOR EDITOR AND REVIEWERS: data and code will be made publicly available upon acceptance.
During review, data and code can be downloaded from the following link  
https://www.dropbox.com/sh/e7r0xn0onkrhu4w/AADirFYTp5iHYzL4YIXnQ92Za?dl=0. \*\*\*

## Results

### Taxonomic survey

```{r top-taxa}
# Count the number of species per 
# family and genus at Nectandra
family_count <-
  nectandra_specimens_strict %>%
  left_join(ppgi, by = "genus") %>%
  assert(not_na, family, genus, species) %>%
  select(family, genus, species) %>%
  unique %>%
  group_by(family) %>%
  summarize(n = n()) %>%
  arrange(desc(n))

genus_count <-
  nectandra_specimens_strict %>%
  left_join(ppgi, by = "genus") %>%
  assert(not_na, family, genus, species) %>%
  select(family, genus, species) %>%
  unique %>%
  group_by(genus) %>%
  summarize(n = n()) %>%
  arrange(desc(n))

# Obtain asymptotic estimated species richness
est_rich <-
richness_estimate[["AsyEst"]] %>%
  as.data.frame %>%
  rownames_to_column("metric") %>%
  as_tibble %>%
  janitor::clean_names() %>%
  filter(metric == "Species Richness") %>%
  mutate_if(is.numeric, round)

total_taxa <- n_distinct(nectandra_specimens_strict$taxon)

# Summarize number of taxa in each class
class_data <- nectandra_specimens_strict %>%
  assert(not_na, genus) %>%
  left_join(ppgi, by = "genus") %>%
  assert(not_na, class, genus, species) %>%
  select(species, class) %>%
  unique() %>%
  count(class) %>%
  mutate(percent = n %>% magrittr::divide_by(total_taxa) %>% scales::percent(accuracy = 0.1)) %>%
  mutate(n = map_chr(n, ~si(., "spp."))) %>%
  mutate(label = glue::glue("{n} ({percent})"))
  
n_ferns <- class_data %>% filter(class == "Polypodiopsida") %>% pull(label)
n_lycos <- class_data %>% filter(class == "Lycopodiopsida") %>% pull(label)

# Summarize growth habit data from checklist
habit_data <-
checklist %>%
  select(scientific_name, habit) %>%
  tidyr::separate_rows(habit, sep = ",") %>%
  mutate(habit = str_trim(habit) %>% str_to_lower()) %>%
  # Combine climbing and clambering
  mutate(habit = str_replace_all(habit, "climbing|clambering", "climbing or clambering")) %>%
  count(habit) %>%
  mutate(percent = n %>% magrittr::divide_by(total_taxa) %>% scales::percent(accuracy = 0.1)) %>%
  mutate(label = glue::glue("{n} ({percent})")) %>%
  mutate(label_2 = glue::glue("*n* = {n}; {percent}"))
  
n_epiphytic <- habit_data %>% filter(habit == "epiphytic") %>% pull(label_2)
n_terrestrial <- habit_data %>% filter(habit == "terrestrial") %>% pull(label_2)

# Count number of species with multiple varieties
n_mult_vars <-
nectandra_specimens_strict %>% filter(!is.na(infraspecific_rank)) %>%
  select(species, infraspecific_name) %>%
  unique %>%
  count(species) %>%
  filter(n > 1) %>%
  # Verify that there are exactly 2 species with 2 varieties each
  verify(nrow(.) == 2) %>%
  verify(all(n == 2))
```

Our surveys resulted in `r taxa_count %>% pull(specimen)` individuals representing `r taxa_count %>% pull(species) %>% si("spp.")`, `r taxa_count %>% pull(genus)` genera, and `r taxa_count %>% pull(family)` families of pteridophytes (`r s_table("checklist")`).
Two species included multiple varieties (two each).
`r n_ferns` are ferns and `r n_lycos` are lycophytes.
Most taxa are either epiphytic (`r n_epiphytic`) or terrestrial (`r n_terrestrial`) (`r table("growth-habit")`).
All taxa are native, except for *Macrothelypteris torresiana* (Gaudich.) Ching (native to Africa and Asia; introduced in the Americas), which was excluded from the comparison of richness across sites and DNA barcode analysis.

**`r table_full("growth-habit")`** Count includes taxa at the species and infraspecies (variety or subspecies) levels (*n* = `r total_taxa` total).

\singlespacing

```{r growth-habit, results = "asis"}
# Format growth habit summary as table
habit_data %>%
  mutate(habit = str_to_sentence(habit)) %>%
  # need to escape percent sign by adding a single backslash
  # (that equates to four backslashes here)
  mutate(label = str_replace(label, "%", "\\\\%")) %>%
  column_to_rownames("habit") %>%
  select(`\\textit{n} (percent of total\\textsuperscript{a})` = label) %>%
  knitr::kable(format = "latex", booktabs = TRUE, escape = FALSE) %>%
  kableExtra::kable_styling(latex_options = "striped")
```

^a^Percentages do not sum to 100% because some taxa have multiple growth habits.

\doublespacing

The genera with the most species were *`r top_taxon(genus_count, 1, genus)`* (`r top_taxon_rich(genus_count, 1)`), *`r top_taxon(genus_count, 2, genus)`* (`r top_taxon_rich(genus_count, 2)`), *`r top_taxon(genus_count, 3, genus)`* (`r top_taxon_rich(genus_count, 3)`), and *`r top_taxon(genus_count, 4, genus)`* (`r top_taxon_rich(genus_count, 4)`).
Families with the most species were `r top_taxon(family_count, 1, family)` (`r top_taxon_rich(family_count, 1)`), `r top_taxon(family_count, 2, family)` (`r top_taxon_rich(family_count, 2)`), and `r top_taxon(family_count, 3, family)` (`r top_taxon_rich(family_count, 3)`).

Three taxa could not be matched to any known species: *Polyphlebium* sp1 (*Nitta 123* and *Nitta 2378*), *Campyloneurum* sp1 (*Nitta 2308*), and *Megalastrum* sp1 (*Nitta 727*).

```{r top-sites}
# Define helper functions to extract sites with highest species richness

# - shorten a site name to just "Nectandra"
short_nec_name <- function (x) {
  if(str_detect(x, "Nectandra")) return ("Nectandra")
  x
}

# Rank sites by a criterion (rank_by) 
# and select the top 'n' site (rank)
top_site_name <- function (df, rank, rank_by) {
  
  rank_by <- enquo(rank_by)
  
  df %>% 
    arrange(desc(!!rank_by)) %>% 
    slice(rank) %>% 
    pull(full_name) %>% 
    short_nec_name
}

# Rank sites by a criterion (rank_by) 
# and return the value of the criterion used for ranking
top_site_num <- function (df, rank, rank_by) {
  
  rank_by <- enquo(rank_by)
  
  df %>% 
    arrange(desc(!!rank_by)) %>% 
    slice(rank) %>% 
    pull(!!rank_by)
}


# Obtain the site names and species richness of the top three most pteridophyte-rich
# protected areas in Costa Rica
top_1_site_name <- top_site_name(cr_richness, 1, richness)
top_1_site_rich <- top_site_num(cr_richness, 1, richness) %>% si("spp.")
top_2_site_name <- top_site_name(cr_richness, 2, richness)
top_2_site_rich <- top_site_num(cr_richness, 2, richness) %>% si("spp.")
top_3_site_name <- top_site_name(cr_richness, 3, richness)
top_3_site_rich <- top_site_num(cr_richness, 3, richness) %>% si("spp.")

# Also do top two by species per hectare
top_1_sp_per_hec_site_name <- top_site_name(cr_richness, 1, richness_per_ha)
top_1_sp_per_hec <- top_site_num(cr_richness, 1, richness_per_ha) %>% round_t(2)
top_2_sp_per_hec_site_name <- top_site_name(cr_richness, 2, richness_per_ha)
top_2_sp_per_hec <- top_site_num(cr_richness, 2, richness_per_ha) %>% round_t(2)
```

Our literature survey identified five other protected areas in mainland Costa Rica that have been surveyed for pteridophytes [@Bigelow1991; @oet2019; @Kappelle2000; @Rojas2008; @Jimenez2016] (`r figure("map")`, `r table("richness-cr")`).
The site with the highest species richness is `r top_1_site_name` (`r top_1_site_rich`), with `r top_3_site_name` (`r top_3_site_rich`) the third-highest after `r top_2_site_name` (`r top_2_site_rich`; `r table("richness-cr")`).
`r top_1_sp_per_hec_site_name` has by far the most species per hectare (`r top_1_sp_per_hec`), with `r top_2_sp_per_hec_site_name` second-highest (`r top_2_sp_per_hec` species per hectare; `r table("richness-cr")`).

**`r table_full("richness-cr")`** A single non-native species, *Macrothelypteris torresiana*, was excluded from calculations for Nectandra.

\singlespacing

```{r richness-cr, results = "asis"}
# Format and output richness table
# FIXME: Be sure to check references numbers to make sure they match final version!
cr_richness %>%
  mutate_at(vars(min_el_m, max_el_m), ~scales::number(., big.mark = ",")) %>%
  mutate(elevation = case_when(
    !is.na(min_el_m) & !is.na(max_el_m) ~ glue::glue("{min_el_m}–{max_el_m}"),
    is.na(min_el_m) ~ glue::glue("ca. {max_el_m}"),
    is.na(max_el_m) ~ glue::glue("ca. {min_el_m}")
  )) %>%
  mutate(holdridge_type = str_to_sentence(holdridge_type)) %>%
  mutate_at(vars(area_ha, richness), ~scales::number(., big.mark = ",")) %>%
  mutate(richness_per_ha = jntools::round_t(richness_per_ha, 2)) %>%
  mutate(citation = str_remove_all(citation, "@")) %>%
  arrange(full_name) %>%
  select(full_name, 
         `Elevation (m)` = elevation, 
         `Life zone type\\textsuperscript{a}` = holdridge_type, 
         `Area (ha)` = area_ha, 
         `Richness (no. spp.)` = richness, 
         `Richness per area (no. spp. per ha)` = richness_per_ha, 
         Reference = citation_number) %>%
  column_to_rownames("full_name") %>%
  knitr::kable(format = "latex", booktabs = TRUE, escape = FALSE) %>%
  kableExtra::kable_styling(latex_options = "striped") %>%
  kableExtra::row_spec(0, bold = TRUE) %>%
  kableExtra::column_spec(1, width = "8em") %>%
  kableExtra::column_spec(3, width = "8em") %>%
  kableExtra::column_spec(5, width = "6em") %>%
  kableExtra::column_spec(6, width = "6em")
```

^a^Life zones follow Holdridge [@Holdridge1967].

\doublespacing

The collection curve did not reach an asymptote (`r figure("richness-inext")`).
Extrapolation of the curve indicates that asymptotic species richness may approach `r si(est_rich$estimator, "spp.")` (95% confidence interval `r si(est_rich$x95_percent_lower, "spp.")` to `r si(est_rich$x95_percent_upper, "spp.")`).

**`r figure_full("richness-inext")`** 
Point at transition from solid to dashed line indicates observed richness. 
Gray shading indicates 95% confidence interval.

### Phylogenetic analysis

```{r seq-stats}
# Make table of newly generated sequences that will be submitted to GenBank
new_seqs_data <-
tibble(genomic_id = names(nectandra_rbcL_raw)) %>%
  left_join(nectandra_dna, by = "genomic_id") %>%
  left_join(nectandra_specimens, by = "specimen_id") %>%
  assert(not_na, taxon)

# - Check number of new rbcL sequences: by taxa
n_taxa_rbcL_new <-
new_seqs_data %>% pull(taxon) %>% n_distinct()

# - Check number of new rbcL sequences: by accession
n_acc_rbcL_new <-
new_seqs_data %>% pull(genomic_id) %>% n_distinct()

# - Get vector of newly generated sequence lengths
seq_lengths_rbcL_new <-
nectandra_rbcL_raw %>% map_dbl(length)

# Check number of rbcL taxa including GenBank
n_taxa_rbcL_seqs <-
  # Start from complete DNA alignment
  tibble(tip = rownames(nectandra_rbcL)) %>%
  # Separate out taxa name by dropping specimen
  separate(tip, into = c("taxon", "specimen"), sep = "_Nitta|_Moran|_Kessler") %>%
  mutate(taxon = str_replace_all(taxon, "_", " ")) %>%
  # Check that worked as expected
  assert(in_set(nectandra_specimens$taxon), taxon) %>%
  # Count number of distinct taxa
  pull(taxon) %>%
  n_distinct()

# Check coverage of rbcL taxa including GenBank
coverage_taxa_rbcL_seqs <- 
  n_taxa_rbcL_seqs %>%
  magrittr::divide_by(taxa_count$taxon) %>%
  scales::percent(accuracy = 0.1)

# Extract relevant lines from iqtree log
align_stats <- 
  magrittr::extract(
    iqtree_log, 
    iqtree_log %>% str_detect("Alignment has [:digit:]+ sequences"))

model_stats <-
  magrittr::extract(
    iqtree_log, 
    iqtree_log %>% str_detect("Best-fit model: "))

best_mod <-
  model_stats %>%
  str_match("model: (.+) chosen") %>%
  magrittr::extract(,2)

best_mod_basis <-
  model_stats %>%
  str_match("according to (.+)$") %>%
  magrittr::extract(,2)

# In this case, the same model was selected by all three selection criteria. Verify this is so.
best_mod_aic <- 
  magrittr::extract(
    iqtree_log, 
    iqtree_log %>% str_detect("^Akaike Information Criterion:")) %>%
  str_match("^Akaike Information Criterion:(.+)$") %>%
  magrittr::extract(,2) %>%
  str_trim()

best_mod_aicc <- 
  magrittr::extract(
    iqtree_log, 
    iqtree_log %>% str_detect("^Corrected Akaike Information Criterion:")) %>%
  str_match("^Corrected Akaike Information Criterion:(.+)$") %>%
  magrittr::extract(,2) %>%
  str_trim()

best_mod_bic <- 
  magrittr::extract(
    iqtree_log, 
    iqtree_log %>% str_detect("^Bayesian Information Criterion:")) %>%
  str_match("^Bayesian Information Criterion:(.+)$") %>%
  magrittr::extract(,2) %>%
  str_trim()
  
assertthat::assert_that(isTRUE(all.equal(best_mod_aic, best_mod_aicc, best_mod_bic)))

# Re-root tree to extract support values
rbcL_tree_rooted <- root_on_lycos(rbcL_tree, ppgi)

# Get sequence length of Osmunda regalis rbcL
osmunda_raw_length <-
nectandra_rbcL_raw_renamed["Osmunda_regalis_Nitta_1875"] %>%
  magrittr::extract2(1) %>%
  length()

```

We generated `r n_acc_rbcL_new` new *rbcL* sequences of pteridophytes occurring at Nectandra, representing `r n_taxa_rbcL_new` taxa (`r s_table("genbank")`).
The newly generated sequences are all samples from Nectandra except for *Abrodictyum rigidum* (Sw.) Ebihara & Dubuisson and *Hymenophyllum trapezoidale* Liebm. (`r s_table("genbank")`).
Mean length of the newly generated sequences was `r mean(seq_lengths_rbcL_new) %>% si("bp", accuracy = 1)` ± SD `r sd(seq_lengths_rbcL_new) %>% si("bp", accuracy = 1)`.
*rbcL* could not be sequenced for `r nectandra_rbcL_missing_taxa %>% n_distinct %>% english2` taxa, and only partial sequences (ca. 3' or 5' half of *rbcL*) could be obtained for `r seq_lengths_rbcL_new[seq_lengths_rbcL_new < 1000] %>% length() %>% english2` taxa.
Difficulty in sequencing of these samples may be due to degraded condition of DNA.
Of the taxa that we were unable to newly sequence, sequences of two taxa from specimens from Nectandra and two taxa from specimens from other areas were available on GenBank (`r s_table("genbank")`). 
In total, our *rbcL* sampling included `r n_taxa_rbcL_seqs` taxa (`r coverage_taxa_rbcL_seqs`) of pteridophytes occurring at Nectandra.

The *rbcL* alignment was `r get_number(align_stats, "columns") %>% si("bp")` long including `r get_number(align_stats, "informative sites") %>% si` parsimony-informative sites and `r get_number(align_stats, "sequences") %>% si` sequences.
The `r best_mod` model was selected for ML phylogenetic analysis according to `r best_mod_basis` (the same model was also selected by AIC and corrected AIC).

The *rbcL* phylogeny (`r s_figure("tree")`) was in generally good agreement with recently published plastid phylogenies [@Pryer2004a; @Schuettpelz2007; @Lehtonen2011; @Testo2016a] at the family level and above, and families *sensu* PPGI [@PteridophytePhylogenyGroupI2016] were monophyletic.
One exception was Osmundaceae sister to Gleicheniaceae (`r get_node_support(rbcL_tree_rooted, c("Osmunda_regalis", "Sticherus_retroflexus"))`), which may be due to poor sampling; we could obtain only a `r si(osmunda_raw_length, "bp")` *rbcL* fragment from the sole osmundaceous species at our study site.
Some internal nodes had weak support (\eg, Cyatheales [`r get_node_support(rbcL_tree_rooted, c("Dicksonia_sellowiana", "Cyathea_pinnula"))`], Polypodiidae [leptosporangiates; `r get_node_support(rbcL_tree_rooted, c("Osmunda_regalis", "Lellingeria_melanotrichia"))`]).
It should be noted that IQ-TREE UFboot support values are not analogous to traditional bootstrap values; only nodes receiving SH-aLRT >=80% and UFboot >=95% should be considered reliable [@Minh2013].
As the purpose of this study was not to robustly infer phylogeny across the ferns and lycophytes, we do not discuss such deep relationships further.

A small number of genera were found to be non-monophyletic but lacked strong support.
*Sphaeropteris brunei* (H. Christ) R.M. Tryon is nested within *Cyathea* as the sister to *Cyathea bicrenata* Liebm. (`r get_node_support(rbcL_tree_rooted, c("Sphaeropteris_brunei", "Cyathea_bicrenata"))`). 
*Stenogrammitis limula* (Christ) Labiak is nested within *Lellingeria* as the sister to *Lellingeria hombersleyi* (Maxon) A.R. Sm. (`r get_node_support(rbcL_tree_rooted, c("Lellingeria_hombersleyi", "Stenogrammitis_limula"))`).
These were resolved in their expected genera in broadly sampled trees using all available *rbcL* sequences on GenBank (`r s_figure("cyatheaceae-tree")` and `r s_figure("grammitid-tree")`, respectively), so their irregular placement in the Nectandra *rbcL* tree appears to be an artifact of low sampling rather than poor sequence quality or misidentification.

Sampling of multiple specimens per species for taxa that are morphologically difficult to distinguish revealed several non-monophyletic species, indicative of species complexes (`r s_figure("tree")`).
*Megalastrum apicale* R.C. Moran & J. Prado, *M. atrogriseum* (C. Chr.) A.R. Sm. & R.C. Moran, and *M. longipilosum* A. Rojas are closely related (`r get_node_support(rbcL_tree_rooted, c("Megalastrum_longipilosum_Nitta_2384", "Megalastrum_atrogriseum_Nitta_2385", "Megalastrum_apicale_Nitta_2230"))`), and the monophyly of each lacks support.
*Diplazium carnosum* Christ is non-monophyletic with respect to *D. urticifolium* Christ and *D. macrophyllum* Desv., but this and most other relationships within *Diplazium* lacked support.
*Didymoglossum ekmanii* (Wess. Boer) Ebihara & Dubuisson is nested within, and very closely related to *D. kapplerianum* (J.W. Sturm) Ebihara & Dubuisson (`r get_node_support(rbcL_tree_rooted, c("Didymoglossum_kapplerianum_Nitta_139", "Didymoglossum_ekmanii_Nitta_126", "Didymoglossum_kapplerianum_Nitta_2299A"))`).

### Barcode analysis

The pteridophyte flora of Nectandra includes `r barcode_fail_num(min_distance_table, "CR")` taxa (`r barcode_fail_percent(min_distance_table, "CR")`) that share identical *rbcL* sequences with at least one other taxon.
This failure rate is higher than that of the pteridophytes of Moorea and Tahiti (`r barcode_fail_percent(min_distance_table, "FP")`), but lower than Japan (`r barcode_fail_percent(min_distance_table, "JA")`; `r figure("min-dist")`).

**`r figure_full("min-dist")`** 
(a) Nectandra, (b) Moorea and Tahiti, (c) Japan (all species), and (d) Japan (sexual diploids only).
Red bar indicates interspecific distance of zero, i.e., species that cannot be distinguished using *rbcL*.

## Discussion

Here, we present to our knowledge the first combined taxonomic and molecular survey of pteridophytes of a protected area in Costa Rica.
We place our results in a regional and global context by comparing this flora with other protected areas in Costa Rica and two other sites that have been the focus of DNA barcoding: Tahiti and Japan.

### Taxonomic diversity

Nectandra has the third-highest species richness of protected areas in Costa Rica with data available for ferns and lycophytes, and by far the greatest number of species per hectare (`r table("richness-cr")`).
While the number of species per hectare is not a fair measure of biodiversity per se as the species-area curve is not linear [@Arrhenius1921], it is useful to assess the effectiveness of a given protected area.
Clearly, Nectandra is highly effective at protecting a large number of pteridophyte species given its area.
Furthermore, the collection curve indicates that additional, unsampled species may be present (`r figure("richness-inext")`), adding to the value of this conservation area.
A recent survey of the bryophytes of Nectandra found a similar number of species (188) and also suggested unsampled species remained due to the shape of the collection curve [@Norris2017].

One reason for the high species richness at Nectandra may be its elevation (`r si(1000, "m")` to `r si(1200, "m")`).
Species richness of pteridophytes in the tropics generally reaches a maximum at mid-elevations on mountains [@Kessler2011a], and plot-based surveys of pteridophytes along elevational gradients in Costa Rica spanning \circa{ }`r si(100, "m")` to `r si(3000, "m")` have found maximum richness at `r si(1000, "m")` to `r si(1200, "m")` [@WatkinsJr.2006a; @Karger2011].
Another possible explanation is the presence of secondary forest and reserve edges, which may contribute additional species that would otherwise not occur in primary forest at this elevation (\ie, the "edge effect" [@Leopold1933]).
Nectandra has a high edge to area ratio due to its small size, and edge effects have been demonstrated for pteridophytes in Mexican montane forests [@Silva2018]. 
While we did not collect data to specifically test this hypothesis, the number of taxa restricted to edges at Nectandra is probably no greater than \circa{ }10 (J. Nitta, pers. obs.), so we believe the effect of elevation is likely more important.

### Unidentified taxa and species complexes

Of the unidentified taxa, two are likely hybrids between distinct species.
*Polyphlebium* sp1 (*Nitta 123* and *Nitta 2378*) shares very similar *rbcL* sequences with *Polyphlebium capillaceum*  (L.) Ebihara & Dubuisson but differs from this species by having expanded laminae (vs. laminae reduced to a few cells on either side of the veins) and growing on rocks in stream beds (vs. growing epiphytically on tree ferns).
*Polyphlebium* sp1 has 32 spores per sporangium, a condition that often indicates asexual reproduction via apogamy [@Braithwaite1964].
This may allow it to reproduce by spores despite not being able to complete normal meiosis.
To our knowledge, a count of 32 spores per sporangium has not been previously reported from *Polyphlebium*, and more detailed study is needed to confirm the reproductive mode of this taxon.
*Campyloneurum* sp1 (*Nitta 2308*) matches in *rbcL* exactly with *Campyloneurum angustifolium* (Sw.) Fée, but has narrower fronds.
*Campyloneurum* sp1 has clear, misshappen spores, whereas *C. angustifolium* is a sexual diploid (2*n* = 72) with oblong spores [@Leon1992].
The chloroplast is generally maternally inherited in ferns [@Gastony1992; @Vogel1998; @Kuo2018a].
Therefore, it is likely that each of these taxa is a hybrid between the species with which they share *rbcL* as the mother and another unknown species as the father.

*Megalastrum* sp1 (*Nitta 727*) shares common aspects of morphology with *M. longipilosum*, but the *rbcL* sequences of *M. apicale*, *M. atrogriseum*, and *M. longipilosum* are extremely similar.
This group is likely a species complex and needs further study, in particular with regards to ploidal level and reproductive mode.
Similarly, *Diplazium* also showed a high degree of morphological diversity but extremely low divergence in *rbcL* sequences, and at least one species, *D. carnosum*, appears to be non-monophyletic.
*Diplazium* at Nectandra may also be a species complex including multiple hybrids and parental taxa.
Sequencing of nuclear genes is needed to clarify relationships for both the putative hybrids (*Polyphlebium* sp1 and *Campyloneurum* sp1) as well as these putative species complexes.

### Molecular diversity and DNA barcode suitability

Unlike animals, there is no single DNA barcode available for plants that works to reliably identify species across all taxonomic groups [@Group2008].
*rbcL* has relatively high phylogenetic informativeness in ferns and lycophytes at the species level and is one of the most frequently sequenced plastid genes in molecular systematic studies.
In previous comparisons with other common plant barcode markers in pteridophytes, *rbcL* performs better than *trnH-psbA* and comparably with *matK* [@Li2011; @Ebihara2010; @Nitta2017].
However, universal primers for *matK* in pteridophytes are lacking [@Li2011].
*rbcL* therefore is a reasonable choice for a single barcode marker in pteridophytes.
However, there have been relatively few studies comparing the performance of *rbcL* as a barcode marker across different pteridophyte floras.

We find that the pteridophytes of Nectandra have a species identification failure rate intermediate between that of Moorea and Tahiti (French Polynesia) and Japan (`r figure("min-dist")`).
The distribution of minimum interspecific distances may reflect the biogeographic history of each region.
Costa Rica and Japan are both mainland areas (or formerly connected with the mainland in the case of Japan), whereas the islands of French Polynesia are extremely isolated and have never been connected to a continent.
It is likely that more species in French Polynesia are recent immigrants that have mostly evolved elsewhere, compared to more species that have evolved *in situ* in Costa Rica and Japan.
This would result in the observed distribution of high interspecific divergences for species from French Polynesia and low interspecific divergences for species in Costa Rica and Japan.

Furthermore, the high failure rate in Japan is likely due to a high rate of apogamy combined with the taxonomic practice of splitting apogamous and sexual forms into separate species [@Ebihara2019b].
When only sexual diploids are included, the failure rate in Japan drops to `r barcode_fail_percent(min_distance_table, "JAsexdip")` and the distribution of minimum interspecific distances more closely resembles that of Nectandra (`r figure("min-dist")`).

## Conclusions

The Nectandra Cloud Forest Reserve is clearly an important site for biodiversity of ferns and lycophytes in Costa Rica, harboring a high number of species for its relatively small area.
Our surveys have already revealed multiple taxa that appear to be new to science, and more likely remain to be discovered.
We demonstrated that *rbcL* can be used to reliably distinguish species in \circa{ }`r barcode_fail_percent(min_distance_table, "CR") %>% parse_number %>% magrittr::subtract(100, .)`% of cases at Nectandra.
This will enable future studies to investigate the ecology of neotropical pteridophytes in more depth, particularly with regards to the gametophytic phase.
Other studies using *rbcL* as a DNA barcode to identify fern gametophytes to species have revealed particular clades and morphologies that tend to occur long distances from conspecific sporophytes [@Ebihara2013; @Nitta2017], but such patterns have yet to be observed in the Neotropics.
We hope our study will lead to more molecular ecological research incorporating field observations of gametophytes and contribute to the systematics of neotropical pteridophytes.

## Acknowledgments

The authors are grateful to Evelyne and David Lennette for allowing us to study the pteridophytes of Nectandra over multiple field seasons and for providing logistical and financial support.
Freddy Castillo Arroyo assisted with fieldwork.
This study was supported by the Japan Society for the Promotion of Science (Kakenhi grant no. 15K07204).
This study is dedicated in memory of Álvaro Ulgade, father of the Costa Rican national park system, friend of Nectandra, and an inspiration to all those who seek to understand and protect biodiversity.

## References
<!-- Insert <div id="refs"></div> to place the refs here instead of at the end -->
<!-- FIXME: remove extra space before each reference in MS Word-->
<div id="refs"></div>

\clearpage

## Supporting information

**`r s_table_full("genbank")`** All specimens from the Nectandra Cloud Forest Reserve, Alajuela Province, Costa Rica, except for *Abrodictyum rigidum* (Sw.) Ebihara & Dubuisson (*Nitta 2012*, Palmira, Alajuela Province, Costa Rica), *Hymenophyllum trapezoidale* Liebm. (*Nitta 169*, San Gerardo de Dota, San José Province, Costa Rica), *Radiovittaria remota* (Fée) E.H. Crane (*R. Moran 3180a*, Costa Rica), and *Trichomanes polypodiodes* L. (*M. Kessler 8808*, Bolivia).

**`r s_table_full("checklist")`** All voucher specimens deposited in UC with duplicates when available at CR, GH, and TNS. For additional data on each voucher specimen, see Dryad repository [@Nitta2020b].

**`r s_figure_full("tree")`** Tree rooted on lycophytes.
Numbers at nodes indicate SH-aLRT support (%) / UFboot support (%); values less than 50% shown with "-"; values of 100% shown with "\*".
For phylogram on right side, scale bar shows expected number of changes per site.
Numbers after species name are J. H. Nitta specimen collection numbers except for *Trichomanes polypodiodes* (*M. Kessler 8808*, Bolivia) and *Radiovittaria remota* (*R. Moran 3180a*, Costa Rica), which could not be sequenced successfully, so GenBank sequences were used instead (accessions AY175795 and U21289, respectively). 
All J. H. Nitta specimens from Nectandra except for *Abrodictyum rigidum* (*Nitta 2012*, Palmira, Alajuela Province, Costa Rica) and *Hymenophyllum trapezoidale* (*Nitta 169*, San Gerardo de Dota, San José Province, Costa Rica).

**`r s_figure_full("cyatheaceae-tree")`** Tree rooted on species from Nectandra in family Dicksoniaceae.
Numbers at nodes indicate local support values computed with the Shimodaira-Hasegawa test; values less than 50% not shown.
For phylogram on right side, scale bar shows expected number of changes per site.
Numbers after species names are GenBank accession numbers for sequences downloaded from GenBank or J. H. Nitta specimen collection numbers for sequences newly obtained by this study. 
Newly obtained ingroup sequences highlighted in yellow.

**`r s_figure_full("grammitid-tree")`** Tree rooted on species from Nectandra in subfamily Polypodioideae.
Numbers at nodes indicate local support values computed with the Shimodaira-Hasegawa test; values less than 50% not shown.
For phylogram on right side, scale bar shows expected number of changes per site.
Numbers after species names are GenBank accession numbers for sequences downloaded from GenBank or J. H. Nitta specimen collection numbers for sequences newly obtained by this study. 
Newly obtained ingroup sequences highlighted in yellow.

\clearpage

```{r map, fig.height = 7.25, fig.width = 7.5, fig.cap = figure_cap("map")}
# Make Costa Rica map showing location of conservation areas

# First add column to map data for whether site is in the "central"
# part of the country (used to color lines when repelling labels)
cr_richness_for_map <-
cr_richness %>%
  mutate(site_type = case_when(
    name %in% c("Monteverde", "San Luis", "Alberto Brenes", "Nectandra") ~ "central",
    TRUE ~ "outer"
  ))

costa_rica_map <- ggplot() +
  # costa_rica_el is elevation data with lat, long, and altitude (in m)
  geom_raster(data = costa_rica_el, aes(lon, lat, fill = alt)) +
  scale_fill_scico(palette = "lajolla", end = 0.9) +
  # min.segment.length isn't an aesthetic, but segment.color is.
  # So we can use segment.color to hide lines that don't need to be plotted.
  geom_text_repel(
    data = cr_richness_for_map,
    aes(x = longitude, y = latitude, label = name, segment.color = site_type),
    size = 3, # text size
    seed = 9130,
    min.segment.length = 0,
    color = "white",     # text color
    bg.color = "grey10", # shadow color
    bg.r = 0.10 # shadow radius
  ) +
  scale_fill_manual(
    aesthetics = "segment.colour",
    values = c(central = "black", other = "transparent"),
    guide = FALSE
    ) +
  geom_point(
    data = cr_richness_for_map %>% filter(name != "Nectandra"), 
    aes(x = longitude, y = latitude), 
    shape = "circle filled", fill = "black", size = 1.5) +
  geom_point(
    data = cr_richness_for_map %>% filter(name == "Nectandra"), 
    aes(x = longitude, y = latitude), 
    shape = "diamond filled", fill = "yellow", size = 2.0) +
  scale_y_continuous(labels = scales::label_number(suffix = "\u00b0N", accuracy = 1)) +
  scale_x_continuous(labels = scales::label_number(suffix = "\u00b0W", accuracy = 1, scale = -1)) +
  labs(
    fill = "Elevation (m)",
    shape = "Site"
  ) +
  coord_quickmap() +
  standard_theme3() +
  theme(
    axis.title.x = element_blank(),
    axis.title.y = element_blank(),
    panel.background = element_rect(fill = "grey80")
  )

# Load Nectandra map created with Inkscape
nectandra_map <- ggdraw() + draw_image(here("images/nectandra_property_map.png"))

# Load primary forest photo
photo <- ggdraw() + draw_image(here("images/P1070534.jpg"))

# Create a blank spacer plot to add some white space
# between Nectandra map and photo
blank <- patchwork::plot_spacer()

# The top part of the plot is just the Costa Rica map
top <- plot_grid(costa_rica_map, labels = "a", label_fontface = "plain")

# The bottom is Nectandra map and photo
bottom <- plot_grid(
  nectandra_map, blank, photo, 
  nrow = 1, 
  labels = c("b", "", "c"), 
  rel_widths = c(54.75, 0.5, 54.75),
  label_fontface = "plain")

# Put them together
plot_grid(top, bottom, nrow = 2, ncol = 1)

# Save the file for submission
ggsave(filename = fig_path("map", ".tiff"), height = 7.25, width = 7.5, units = "in", dpi = 300)
```

\clearpage

```{r richness-inext, fig.height = 3, fig.width = 3, fig.cap = figure_cap("richness-inext")}
# Make species collection curve
make_inext_plot(richness_estimate)

ggsave(filename = fig_path("richness-inext", ".eps"), height = 3, width = 3, units = "in", device = cairo_ps_high_res)
```

\clearpage

```{r min-dist, fig.height = 4.5, fig.width = 7, fig.cap = figure_cap("min-dist")}
# Make barcode distances histogram

# Format min. intersp. distances for plotting:
# add facet labels
min_distance_plot_data <-
min_distance_table %>%
  mutate(
    facet_label = case_when(
    dataset == "CR" ~ "a",
    dataset == "FP" ~ "b",
    dataset == "JA" ~ "c",
    dataset == "JAsexdip" ~ "d"
  ) %>% as.factor()
  )
  
# Make separate df of label data to add
# dataset labels to each subplot
min_distance_label_data <- 
  min_distance_plot_data %>%
  # Make inset labels
  mutate(part_label = case_when(
    dataset == "CR" ~ "Nectandra",
    dataset == "FP" ~ "Moorea and Tahiti",
    dataset == "JA" ~ "Japan",
    dataset == "JAsexdip" ~ "Japan (sex. dip.)"
  )) %>%
  select(facet_label, part_label) %>%
  unique

# Make plot
ggplot(min_distance_plot_data, aes(range, percent, fill = is_zero)) +
  # Main geom: columns
  geom_col() +
  # Add dataset labels in upper-right
  geom_label(
    data = min_distance_label_data,
    aes(label = part_label), 
    x = Inf, y = Inf, 
    hjust = "inward",
    vjust = "inward", 
    fill = "white",
    label.size = 0,
    size = 9/ggplot2::.pt) +
  # Color zero distances red
  scale_fill_manual(values = c("FALSE" = "grey40", "TRUE" = "#de2d26")) +
  # Shrink the auto expand scaling
  scale_x_continuous(labels = scales::percent, expand = c(0.01, 0)) +
  scale_y_continuous(labels = scales::percent) +
  standard_theme3() +
  theme(legend.position = "none") +
  labs(x = "Genetic distance", y = "Relative frequency") +
  theme(
    # Use only thin horizontal gridlines
    panel.grid.minor.x = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.major.y = element_line(size = rel(0.5)),
    # Set facet labels to left side, no box
    strip.text = element_text(hjust = 0),
    strip.background = element_blank()
  ) +
  # Wrap by dataset
  facet_wrap(vars(facet_label), ncol = 1)

ggsave(filename = fig_path("min-dist", ".eps"), height = 4.5, width = 7, units = "in", device = cairo_ps_high_res)
```

\clearpage
